vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -0,0 +1,453 @@
+#include "connection_box_lookahead_map.h"
+
+#include <vector>
+#include <queue>
+
+#include "connection_box.h"
+#include "rr_node.h"
+#include "router_lookahead_map_utils.h"
+#include "globals.h"
+#include "vtr_math.h"
+#include "vtr_time.h"
+#include "echo_files.h"
+
+/* we're profiling routing cost over many tracks for each wire type, so we'll
+ * have many cost entries at each |dx|,|dy| offset. There are many ways to
+ * "boil down" the many costs at each offset to a single entry for a given
+ * (wire type, chan_type) combination we can take the smallest cost, the
+ * average, median, etc. This define selects the method we use.
+ *
+ * See e_representative_entry_method */
+#define REPRESENTATIVE_ENTRY_METHOD SMALLEST
+
+#define REF_X 25
+#define REF_Y 23
```

Comment:
What are these numbers related to? Meaning, do they work well generally or are they tuned for specific cases?

Reply:
They are tuned, but so are the look ahead REF_X/Y, so I'm not super concerned with this at the outset.